internal/plans/planfile/reader.go

```diff
@@ -50,15 +51,27 @@ func (e *ErrUnusableLocalPlan) Unwrap() error {
 // be used to access the individual portions of the file for further
 // processing.
 type Reader struct {
-	zip *zip.ReadCloser
+	zip *zip.Reader
 }
 
 // Open creates a Reader for the file at the given filename, or returns an error
 // if the file doesn't seem to be a planfile. NOTE: Most commands that accept a
 // plan file should use OpenWrapped instead, so they can support both local and
 // cloud plan files.
-func Open(filename string) (*Reader, error) {
-	r, err := zip.OpenReader(filename)
+func Open(filename string, enc encryption.PlanEncryption) (*Reader, error) {
+	raw, err := os.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	println("A")
+	println(raw)
+
+	decrypted, diags := enc.DecryptPlan(raw)
```

Comment:
This is diags for now as go has wierd edge cases about casting a nil object to an error.